### DISCUSSION TITLE
icons: are there restrictions/limits for adding new icons?

### DISCUSSION THREAD

**Discussion by hasecilu** [eza-community/eza#704] [Development]
In #565 I mentioned about the possibility of adding more file extensions for Godot files and now on #680 I face the same issue: some programs have several types of files that result on several file extensions.

-  Godot
There are a lot of ASCII text file extensions used for different script/assets. `.gd`, `tscn`, `tres`.
-  KiCad
Since it's an EDA tool it deals with several file types and use these extensions. `.kicad_mod`, `.kicad_pcb`, `.kicad_prl`, `.kicad_pro`, `.kicad_sch`, `.kicad_sym`
-  FreeCAD
Since it's a CAD tool it deals with several file types (some native and a lot other 3rd party) and use these extensions. `fcmacro`, `fcmat`, `fcparam`, `fcscript`, `fcstd`, `fctb`, `fctl`
-  Krita
It uses 2 extensions for file (`.kra`, `.krz`) and one for brushes (`.kpp`)
-  Kdenlive
Apart from the project, titles formatting (`.kdenlivetitle`)

I'm pretty sure a lot of programs could have a lot of auxiliary/secondary/assets file extensions associated with them so the question is: **is encouraged to add a lot of icons from the same program or is it preferred to only add the main project file?** **Are there guidelines for submitting new icons?**

### Advantage

Files that use these extensions (and more in the future) gets their custom icons instead of the default   icon.

### Disadvantage
It could slow the program. (I don't know how much the performance could be impacted, you tell me)

---
Also in  #565 I mentioned different types of 3D files but I didn't find suitable icons CAD-related on NF so they may be added to some place.

**Comment by PThorpe92** [comment]
Generally adding new icons is definitely encouraged :+1: , as long as that filetype doesn't conflict with an existing one (for instance I think there was a `.v` filetype that conflicted with the V language at some point.. That is where it gets difficult)

But any time an existing one is changed, that has to be approved by everyone so it can take a bit longer because most of us work and getting a few people to all review the same PR can be a hassle. 

I just looked again at the PR with the new Nerd fonts, I think there are some good additions there :+1:  gonna upgrade my font

**Reply by hasecilu** [reply]
Cool, my main concern was to not being overpopulating the maps with a lot of icons.
Still I wait a little bit to update for the missing file extensions in the PR.

**Reply by PThorpe92** [reply]
No thankfully that isn't really a concern :+1:  The main concern would be to make sure that if we add file extension X, that if somehow X has two different programs/systems that use that extension (not super common but it happens), that we choose the icon that makes the most sense.

Thanks for the contribution tho it looks like you added quite a few :heart:

**Reply by hasecilu** [reply]
Yeah, I added a `touch` command for you all to replicate all filenames fast.
